### PR TITLE
Second round of Det64 updates

### PR DESCRIPTION
**Pull request by jeffschriber**
This PR focuses on a few facets of det64 functionality:
 - Further optimization of CI macrofunctions:
      - [x] Vector-based determinant screening

 - Conversion of base functions involving determinants to maximize use of bit-operations
     - [X] Energy function

- Misc.
     - [x] Enable ACISCF-DSRG

- [x] Ready to go

### CODE REVIEW

**Comment by fevangelista** [comment]
@jeffschriber: I see some of the old commits from the previous det64 branch. Could it be because it was not rebased? I assume this is still a work in progress so I will wait for your commits.

**Comment by jeffschriber** [comment]
I just rebased this branch. I think It should be merged once the tests pass.

**Comment by fevangelista** [comment]
I went ahead and rebased. I still need some time to check the code.

**Comment by fevangelista** [comment]
@jeffschriber: can you please check the `aci-dsrg-mrpt2-5` test? It is failing on Travis and on my laptop.

**Comment by jeffschriber** [comment]
Interesting, I'll look in to that test case right away

**Comment by fevangelista** [comment]
Ok, now that it passes I'll start the review.

**Comment by fevangelista** [comment]
@tyzhang1993: is there a way we can undo the merge and rebase with master? While reviewing I see a lot of changes from other commits.

**Comment by jeffschriber** [comment]
Thanks Sam, your procedure is the right one. I'm a little puzzled by what was happening, because I did always use `--rebase` when I merge though I did see commits repeated after that. Perhaps I did something wrong that I'm not noticing.

File: src/aci/aci.cc [hunk 1]
```diff
@@ -3322,12 +3352,28 @@ void AdaptiveCI::spin_analysis() {
...
     std::vector<double> l1b(L1bT.data());
     std::vector<double> l2aa(L2aaT.data());
     std::vector<double> l2ab(L2abT.data());
     std::vector<double> l2bb(L2bbT.data());
+    for (int i = 0; i < nact; ++i) {
+        for (int j = 0; j < nact; ++j) {
+            double value = ( l2aa[i * nact3 + j * nact2 + i * nact + j] +
+                                    l2bb[i * nact3 + j * nact2 + i * nact + j] -
+                                    l2ab[i * nact3 + j * nact2 + i * nact + j] -
+                                    l2ab[j * nact3 + i * nact2 + j * nact + i] );
```
Comment (tyzhang1993) [inline src/aci/aci.cc]: Repeated computation of indices can be avoided by declare an local variable.

**Pull request merged**